tests/networkpolicy_test.go

```diff
@@ -223,17 +287,70 @@ func assertPingFail(fromVmi, toVmi *v1.VirtualMachineInstance) {
 	}, 5*time.Second, 1*time.Second).ShouldNot(Succeed())
 }
 
-func waitForNetworkPolicyDeletion(policy *v1network.NetworkPolicy) {
+func createNetworkPolicy(namespace, name string, labelSelector metav1.LabelSelector, ingress []networkv1.NetworkPolicyIngressRule) *networkv1.NetworkPolicy {
+	policy := &networkv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: networkv1.NetworkPolicySpec{
+			PodSelector: labelSelector,
+			Ingress:     ingress,
+		},
+	}
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	By(fmt.Sprintf("Create networkpolicy %s/%s", policy.Namespace, policy.Name))
+	policy, err = virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Create(policy)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred(), fmt.Sprintf("should succeed creating network policy %s/%s", policy.Namespace, policy.Name))
+	return policy
+}
+
+func waitForNetworkPolicyDeletion(policy *networkv1.NetworkPolicy) {
 	if policy == nil {
 		return
 	}
 
 	virtClient, err := kubecli.GetKubevirtClient()
 	ExpectWithOffset(1, err).ToNot(HaveOccurred())
 
-	ExpectWithOffset(1, virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, &v13.DeleteOptions{})).To(Succeed())
+	ExpectWithOffset(1, virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, &metav1.DeleteOptions{})).To(Succeed())
 	EventuallyWithOffset(1, func() error {
-		_, err := virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Get(policy.Name, v13.GetOptions{})
+		_, err := virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Get(policy.Name, metav1.GetOptions{})
 		return err
 	}, 10*time.Second, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())))
 }
+
+func checkHTTPPingFn(fromVmi, toVmi *v1.VirtualMachineInstance, port int) func() error {
+	return func() error {
+		const curlCheckCmd = "curl --head %s --connect-timeout 5\n"
+		for _, ip := range toVmi.Status.Interfaces[0].IPs {
+			if netutils.IsIPv6String(ip) {
+				ip = "[" + ip + "]"
+			}
+			url := fmt.Sprintf("http://%s:%d", ip, port)
+			curlCheck := fmt.Sprintf(curlCheckCmd, url)
+			err := tests.VmiConsoleRunCommand(fromVmi, curlCheck, 10*time.Second)
+			if err != nil {
+				return fmt.Errorf("failed HTTP ping from vmi(%s/%s) to url(%s) at vmi(%s/%s): %v", fromVmi.Namespace, fromVmi.Name, url, toVmi.Namespace, toVmi.Name, err)
+			}
+		}
+		return nil
+	}
+}
+
+func assertHTTPPingSucceed(fromVmi, toVmi *v1.VirtualMachineInstance, port int) {
+	EventuallyWithOffset(1, checkHTTPPingFn(fromVmi, toVmi, port), 10*time.Second, time.Second).Should(Succeed())
```

Comment:
no need, just consistently you need on the success
the policy is always pass by default, the positive tests should pass always, just a bug or network problem can cause a packet to fail

Reply:
Done, copy paste issue.